bootstrap_conan_utils.py

```diff
@@ -0,0 +1,565 @@
+#
+# --- Use hash to get download url
+#  curl $APIBASE/conans/pcre/8.45/_/_/packages/139391a944851d9dacf1138cff94b3320d5775dd/download_urls
+#  {
+#    "conaninfo.txt" : "http://localhost:8082/artifactory/api/conan/conan-local/v2/files/_/pcre/8.45/_/125d5f684fea10391ff4cbcd809a5c74/package/139391a944851d9dacf1138cff94b3320d5775dd/ce6f2349e761f6350cbde62b02a687c7/conaninfo.txt",
+#    "conan_package.tgz" : "http://localhost:8082/artifactory/api/conan/conan-local/v2/files/_/pcre/8.45/_/125d5f684fea10391ff4cbcd809a5c74/package/139391a944851d9dacf1138cff94b3320d5775dd/ce6f2349e761f6350cbde62b02a687c7/conan_package.tgz",
+#    "conanmanifest.txt" : "http://localhost:8082/artifactory/api/conan/conan-local/v2/files/_/pcre/8.45/_/125d5f684fea10391ff4cbcd809a5c74/package/139391a944851d9dacf1138cff94b3320d5775dd/ce6f2349e761f6350cbde62b02a687c7/conanmanifest.txt"
+#  }
+#
+# --- Use the url for "conan_package.tgz"
+#  http://localhost:8082/artifactory/api/conan/conan-local/v2/files/_/pcre/8.45/_/125d5f684fea10391ff4cbcd809a5c74/package/139391a944851d9dacf1138cff94b3320d5775dd/ce6f2349e761f6350cbde62b02a687c7/conan_package.tgz
+
+
+def urlenc(s: str) -> str:
+    """Urlencode the given string"""
+    return urllib.parse.quote_plus(s)
+
+def conan_search_package_name(package_name: str, **kwargs) -> json:
+    """
+    Search for the given package name on the conan server
+
+    Parameters:
+      package_name: str The package name to find.  Only the name, no version components.
+    kwargs:
+      repourl: str The repo to search.  Optional, Default DEFAULT_CONAN_URL.
+
+    Returns: json results
+
+    throws Exception on invalid json returned
+    """
+    repourl = kwargs["repourl"] if "repourl" in kwargs else DEFAULT_CONAN_URL
+    queryurl = f"{repourl}/v2/conans/search"
+    headers = {"Content-Type": "application/json"}
+    params = {"q": package_name}
+
+    response = requests.get(queryurl, headers=headers, params=params)
+    return json.loads(response.text)
+
+
+def conan_get_recipe_revisions(package_name: str, package_version: str, **kwargs) -> dict:
+    """
+    Search for the available recipe revisions for the given package name on the conan server
+
+    Parameters:
+        package_name
+        package_version
+    kwargs:
+      repourl: str The repo to search.  Optional, Default DEFAULT_CONAN_URL.
+
+    throws Exception on invalid json returned
+    """
+    repourl = kwargs["repourl"] if "repourl" in kwargs else DEFAULT_CONAN_URL
+    queryurl = f"{repourl}/v2/conans/{urlenc(package_name)}/{urlenc(package_version)}/_/_/revisions"
+    headers = {"Content-Type": "application/json"}
+
+    response = requests.get(queryurl, headers=headers)
+    jresult = json.loads(response.text)
+    if "errors" in jresult:
+      raise Exception("Conan error while getting recipe revisions for \"{package_name}/{package_version}\": \"{jresult}\"")
```

Comment:
In Python it isn't accepted practice to raise `Exception`. The preference is to raise an error like [RuntimeError](https://docs.python.org/3/library/exceptions.html#RuntimeError) in this context.